modules/oe_migration_workflow/tests/src/Unit/Plugin/migrate/process/SetWorkflowStateTest.php

```diff
@@ -0,0 +1,308 @@
+    $workflow_base->expects($this->any())
+      ->method('get')
+      ->with('type_settings')
+      ->willReturn(
+        [
+          'states' =>
+            [
+              'draft' => [
+                'published' => 0,
+              ],
+              'published' => [
+                'published' => 1,
+              ],
+            ],
+        ]
+      );
+
+    $workflow_alternative = $this->createMock(WorkflowInterface::class);
+    $workflow_alternative->expects($this->any())
+      ->method('get')
+      ->with('type_settings')
+      ->willReturn(
+        [
+          'states' =>
+            [
+              'invalid' => [
+                'published' => 0,
+              ],
+              'valid' => [
+                'published' => 1,
+              ],
+            ],
+        ]
+      );
+
+    $configEntityTypeInterface = $this->createMock(ConfigEntityTypeInterface::class);
+    $configEntityTypeInterface->expects($this->any())
+      ->method('getConfigPrefix')
+      ->willReturn('workflows.workflow');
+
+    $storage = $this->createMock(EntityStorageInterface::class);
+    $storage->expects($this->any())
+      ->method('load')
+      ->willReturnMap(
+        [
+          ['oe_corporate_workflow', $workflow_base],
+          ['test', $workflow_alternative],
+        ]
+      );
+
+    $this->entityTypeManager = $this->createMock(EntityTypeManagerInterface::class);
+    $this->entityTypeManager->expects($this->any())
+      ->method('getDefinition')
+      ->with('workflow')
+      ->willReturn($configEntityTypeInterface);
+    $this->entityTypeManager->expects($this->any())
+      ->method('getStorage')
+      ->with('workflow')
+      ->willReturn($storage);
+
+    $this->configManager = $this->createMock(ConfigManagerInterface::class);
+    $this->configManager->expects($this->any())
+      ->method('getEntityTypeIdByName')
+      ->with($this->logicalOr('workflows.workflow.oe_corporate_workflow', 'workflows.workflow.test'))
+      ->willReturn('workflow');
+  }
+
+  /**
+   * Test de base case without any special configuration.
+   *
+   * @throws \Drupal\migrate\MigrateException
+   * @throws \Exception
+   */
+  public function testTransform() {
+    $this->initializePlugin();
+
+    $value = 'published';
+    $statusDestination = $this->plugin->transform($value, $this->migrateExecutable, $this->row, $this->destinationProperty);
+    $this->assertEquals($statusDestination, $value);
+
+    $value = 'draft';
+    $this->row->setSourceProperty('status', 0);
+    $this->row->setSourceProperty('name', $value);
+    $statusDestination = $this->plugin->transform($value, $this->migrateExecutable, $this->row, $this->destinationProperty);
+    $this->assertEquals($statusDestination, $value);
+  }
+
+  /**
+   * Test that unrecognized workflow status are changed with a new valid status.
+   *
+   * @throws \Drupal\migrate\MigrateException
+   * @throws \Exception
+   */
+  public function testTransformNewWorkflowStatus() {
```

Comment:
This can be moved inside the `testTransform` which you can name `testTransformDefault`. You are testing the same workflow but with different values but are expecting the same results due to them being invalid (missing) workflow statuses.

Reply:
mm, yes, ok. Merged and changed the name to testTransformDefault